server/controllers/erasure-search.js

```diff
@@ -0,0 +1,97 @@
+const { default: log } = require('@financial-times/n-logger');
+const operation = 'erasure-search';
+
+async function searchUserTableByUuid (db, uuid, requestedBy) {
+	try {
+		const userTableResults = await db.run(`SELECT * FROM syndication.users WHERE user_id = '${uuid}';`);
+		log.info({ operation, subOp: 'search-user-table-by-uuid', requestedBy, uuid, userTableResults});
+		return { uuid, table: 'users', recordsFound: userTableResults.length };
+	} catch (error) {
+		throw error;
+	}
+}
+
+async function searchUserTableByEmail (db, email, requestedBy) {
+	try {
+		const userTableResults = await db.run(`SELECT * FROM syndication.users WHERE email = '${email}';`);
+		log.info({ operation, subOp: 'search-user-table-by-email', requestedBy, email, userTableResults});
+		return { email, table: 'users', recordsFound: userTableResults.length };
+	} catch (error) {
+		throw error;
+	}
+}
+
+async function searchMigratedUsersTableByUuid (db, uuid, requestedBy) {
+	try {
+		const migratedUserTableResults = await db.run(`SELECT * FROM syndication.migrated_users WHERE user_id = '${uuid}';`);
+		log.info({ operation, subOp: 'search-migrated-user-table-by-uuid', requestedBy, uuid, migratedUserTableResults});
+		return { uuid, table: 'migrated_users', recordsFound: migratedUserTableResults.length };
+	} catch (error) {
+		throw error;
+	}
+}
+
+function getInvalidUuids (uuids) {
+	return uuids.filter(uuid => !/[a-f\d]{8}-[a-f\d]{4}-[a-f\d]{4}-[a-f\d]{4}-[a-f\d]{12}/.test(uuid));
+}
+
+function getInvalidEmails (emails) {
+	return emails.filter(email => !/[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,64}/.test(email));
+}
+
+module.exports = async (req, res, next) => {
+	const requestedBy = req.cookies.s3o_username;
```

Comment:
These responses shouldn't be cacheable by the client or proxies so we need to set `Cache-Control` here.